_posts/2020-02-17-Combining-MongoDB-and-AWS-Lambda.md

```diff
@@ -0,0 +1,310 @@
+---
+layout: post
+authors: [nick_van_hoof]
+title: 'Marrying MongoDB Atlas and AWS Lambda'
+image: /img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/featured-image.png
+tags: [MongoDB, Cloud, AWS, AWS Lambda, Serverless]
+category: Cloud
+comments: false
+---
+
+### Reading time: 10 min 30 sec
+
+# Table of contents
+* [MongoDB and AWS Lambda: a successful marriage?](#mongodb-and-aws-lambda-a-successful-marriage)
+* [MongoDB and AWS Lambda: Why?](#mongodb-and-aws-lambda-why)
+* [MongoDB and AWS Lambda: Performance](#mongodb-and-aws-lambda-performance)
+* [Cold start vs warm performance](#cold-start-vs-warm-performance)
+* [Performance conclusions](#performance-conclusion)
+* [VPC peering between AWS and MongoDB Atlas](#vpc-peering-connect-your-lambda-functions-with-your-mongodb-atlas-cluster)
+* [Useful links](#useful-links)
+
+
+# MongoDB and AWS Lambda: a successful marriage?
+
+<div style="text-align: center;" >
+  <img src="/img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/rings.png" width="15%" height="15%">
+</div>
+
+Can we use MongoDB Atlas when working with AWS Lambda?  
+Yes we can!   
+It's simple enough to setup and above all also performing well!  
+
+This blog concerns mainly two things:
+* Why would you use MongoDB + AWS Lambda and how does it perform
+* How to create a production grade setup with vpc-peering
+
+# MongoDB and AWS Lambda: Why?
+As I am both a fan of AWS Lambda and MongoDB Atlas it was fun for me to marry them.  
+However in the real world we don't do things for fun alone.  
+
+<div style="text-align: center;" >
+  <img src="/img/2020-02-17-Combining-MongoDB-Atlas-and-AWS-Lambda/mongodb-plus-aws-lambda.png" width="70%" height="70%">
+</div>
+
+What are the motives to combine MongoDB Atlas and AWS Lambda?
+
+* In case of MongoDB you provision your cluster and you know what you'll pay for it.
+* MongoDB has a rich language to query within a collection
+* On top of your data in MongoDB you can build rich dashboards for business intelligence.
+* MongoDB supports joins over collections.
+* Support for lot's of indexes
+* Large documents allowed. A MongoDB document can be up to 16 Mb.
+* Flexible data access. Also in a NoSQL world it is important to know your data access model. 
+Data modeling is an important step that needs to be taken carefully before you start persisting your data.
+The way your data is modelled will limit the access patterns that can be used on your data.
+MongoDB offers you more flexibility to access your data in different ways after you have stored it.
+
+# MongoDB and AWS Lambda: Performance
```

Comment:
Add some more explanation about what you are trying to measure with this.
"Performance" is very generic.
When reading this I would like to know upfront what you are going to measure, and for me this is not clear in the beginning .

Reply:
Okay I added a few lines above